apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -1787,6 +1820,82 @@ func (m *ModelManagerAPI) validateNoSeriesUpgrades(st State, force bool) error {
 	return nil
 }
 
+func (m *ModelManagerAPI) ToolVersions(entity params.Entity) (params.ToolVersionsResult, error) {
+	var err error
+	result := params.ToolVersionsResult{}
+
+	modelTag, err := names.ParseModelTag(entity.Tag)
+	if err != nil {
+		return result, errors.Trace(err)
+	}
+	isModelAdmin, err := m.authorizer.HasPermission(permission.AdminAccess, modelTag)
+	if err != nil {
+		return result, errors.Trace(err)
+	}
+	if !isModelAdmin && !m.isAdmin {
+		return result, apiservererrors.ErrPerm
+	}
+
+	st, err := m.statePool.Get(modelTag.Id())
+	if err != nil {
+		return result, errors.Trace(err)
+	}
+	defer st.Release()
+
+	model, err := st.Model()
+	if err != nil {
+		return result, errors.Trace(err)
+	}
+	if !model.IsControllerModel() {
```

Comment:
If this is restricted to the controller model, then there's no need to pass in any model tag as an argument - just use SystemState to get the model.